src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -187,8 +190,14 @@ protected virtual Task WaitForChangesAsync(RequestContext context, CancellationT
                     // Nothing changed between the last request and this one.  Report a (null-diagnostics,
                     // same-result-id) response to the client as that means they should just preserve the current
                     // diagnostics they have for this file.
-                    var previousParams = documentToPreviousDiagnosticParams[diagnosticSource.GetId()];
-                    progress.Report(CreateUnchangedReport(previousParams.TextDocument, previousParams.PreviousResultId));
+                    //
+                    // Note: if this is a workspace request, we can do nothing, as that will be interpreted by the
+                    // client as nothing having been changed for that document.
+                    if (!this.IsWorkspacePullDiagnosticsHandler)
```

Comment:
Alternative approach - allow CreateUnchangedReport to return a nullable value, and in workspace diagnostics return a null value always.

Reply:
fair.  it seemed about the same... i'll noodle on it.